Sources/Shared/Core/VerticalBlueprintLayout.swift

```diff
@@ -159,37 +160,43 @@
         layoutAttribute.size = resolveSizeForItem(at: indexPath)
 
         if let previousItem = previousAttribute {
-          var minY: CGFloat = previousItem.frame.origin.y
-          var maxY: CGFloat = previousItem.frame.maxY
-
-          // Properly align the current item with the previous item at the same
-          // x offset. This helps ensure that the layout renders correctly when
-          // using layout attributes with dynamic height.
-          if let itemsPerRow = itemsPerRow,
-            itemsPerRow > 1,
-            item > Int(itemsPerRow) - 1,
-            Int(itemsPerRow) - 1 < layoutAttributes[section].count {
-            let previousXOffset = item - Int(itemsPerRow - indexOffsetForSectionHeaders(at: sectionIndexPath))
-            let lookupAttribute = layoutAttributes[section][previousXOffset]
-            maxY = lookupAttribute.frame.maxY
-            minY = lookupAttribute.frame.maxY + sectionsMinimumLineSpacing
+          if perRow > 1,
+            perRow > perRow - 1,
+            perRow - 1 < layoutAttributes[section].count {
+            let lookupAttributes = layoutAttributes[section].filter({ $0.representedElementCategory == .cell })
+            if lookupAttributes.count < perRow {
+              layoutAttribute.frame.origin.x = previousItem.frame.maxX + sectionsMinimumInteritemSpacing
+              layoutAttribute.frame.origin.y = previousItem.frame.minY
+            } else {
+              var minimumYAttributes = lookupAttributes.sorted(by: { $0.frame.maxY < $1.frame.maxY })
+              guard let minimumYAttribute = lookupAttributes.filter({ $0.frame.maxY == minimumYAttributes.first!.frame.maxY }).first else {
```

Comment:
Need to decide what we want to do here, I do not like using forced unwrapping or leaving fatalErrors in frameworks. We shouldn't be responsible for the implementers application crashing. However this should be a case that never occurs.

@zenangst - Thoughts?

Reply:
I think we should try and avoid having `fatalError`'s.
I'm working on a commit that would remove the need for it, hang on! :)